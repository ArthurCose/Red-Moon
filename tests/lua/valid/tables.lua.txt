-- saved as txt to avoid formatter

print("basics:")
a = {
  a = 1;
  ["b"] = 2;
  3,
}
print(a.a, a.b, a[1], #a)

print("\nlen outside init:")
a = {}
a[1] = 1
print(#a)

print("\nmerging:")
a = { 1 }
a[3] = 3
print(#a)
a[2] = 2
print(#a)

print("\nmerging in init:")
a = { [2] = 2, 1 }
print(#a)

print("\nshrinking:")
a = { 1, nil, 3 }
print(#a)
a[3] = nil
print(#a)

print("\nvariadic:")
local function foo(...) return ... end
a = { foo(1, 2, 3, 4) }
print(#a, a[2])

print("\nexpresion execution order:")
a = {
  [print(1) or 1] = print(2),
  print(3),
  [print(4) or 1] = print(5),
}

print("\n1 element - late flush:")
a = {
  [1] = 1,
  2,
  [1] = 3
}
print(a[1])

print("\n50 elements - early flush:")
a = {
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  [1] = 2
}
print(#a, a[1])

print("\n49 elements - late flush:")
a = {
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1,
  [1] = 2
}
print(#a, a[1])

print("\nexpression access")
a.a = 2
print((a)[1])
print((a).a)
