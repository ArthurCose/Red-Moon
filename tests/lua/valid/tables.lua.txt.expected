basics:
1	2	3	1

len outside init:
1

merging:
1
3

merging in init:
2

shrinking:
3
1

variadic:
4	2
5	1
4	2
5	1

expresion execution order:
1
2
3
4
5

1 element - late flush:
2

50 elements - early flush:
50	2

49 elements - late flush:
49	1

51 elements - double flush:
51	2	1

preserving outside of flush:
4	7

expression access:
1
3
